### PR TITLE
Don't run full hostname scan when gathering aggregated MTA-STS stats

### PR DESCRIPTION
**Pull request by vbrown608**

### CODE REVIEW

File: db/scripts/init_tables.sql [hunk 1]
```diff
@@ -81,11 +81,12 @@ CREATE TABLE IF NOT EXISTS domain_totals
     time            TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
     source          TEXT NOT NULL,
     attempted       INTEGER DEFAULT 0,
-    connected       INTEGER DEFAULT 0,
     mta_sts_testing INTEGER DEFAULT 0,
```
Comment (sydneyli) [inline db/scripts/init_tables.sql]:
```suggestion
    with_mxs         INTEGER DEFAULT 0,
    mta_sts_testing INTEGER DEFAULT 0,
```
add the column here too, for reference

File: db/scripts/init_tables.sql [hunk 2]
```diff
@@ -81,11 +81,12 @@ CREATE TABLE IF NOT EXISTS domain_totals
...
     mta_sts_enforce INTEGER DEFAULT 0
 );
 
 ALTER TABLE domains ADD COLUMN IF NOT EXISTS queue_weeks INTEGER DEFAULT 4;
 
 ALTER TABLE domains ADD COLUMN IF NOT EXISTS testing_start TIMESTAMP;
+
+ALTER TABLE domain_totals ADD COLUMN IF NOT EXISTS with_mxs INTEGER DEFAULT 0;
```
Comment (sydneyli) [inline db/scripts/init_tables.sql]: wdyt about adding `DROP COLUMN IF EXISTS connected` for the migration?

File: checker/hostname.go [hunk 3]
```diff
@@ -210,29 +210,40 @@ func checkTLSVersion(client *smtp.Client, hostname string, timeout time.Duration
...
 }
 
-// CheckHostname performs a series of checks against a hostname for an email domain.
-// `domain` is the mail domain that this server serves email for.
-// `hostname` is the hostname for this server.
-func (c *Checker) CheckHostname(domain string, hostname string) HostnameResult {
-	if c.checkHostnameOverride != nil {
-		// Allow the Checker to mock this function.
-		return c.checkHostnameOverride(domain, hostname)
+func NoopCheckHostname(domain string, hostname string, _ time.Duration) HostnameResult {
```
Comment (sydneyli) [inline checker/hostname.go]:
```suggestion
// NoopCheckHostname returns a fake error result containing `domain` and `hostname`.
func NoopCheckHostname(domain string, hostname string, _ time.Duration) HostnameResult {
```
oor something like that? to appease the linter

Comment (vbrown608) [reply checker/hostname.go]: Yess thank you for linting my code :)

File: db/scripts/init_tables.sql (hunk 2 above)
Comment (vbrown608) [reply db/scripts/init_tables.sql]: Sure!

**Pull request merged**